### PR TITLE
make a db interface so we can remove web chunks from indexing/populating code

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: symbol_exporter/db_access_model.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            f"{self.host}{url}",
+            data=dumped_data,
+            headers=self._setup_headers(dumped_data, url=url),
+        )
+        r.raise_for_status()
+
+    def get_current_symbol_table_artifacts(self):
+        all_indexted_pkgs = set()
+        symbol_table_url = f"/api/v{version}/symbol_table"
+        extracted_symbols = requests.get(f"{self.host}{symbol_table_url}").json()
```
Comment (dacamo76) [inline symbol_exporter/db_access_model.py:51]: `host` is not an instance variable, no need to make this an instance method, would make it easier to parallelize/test if extracted as separate function.

File: symbol_exporter/db_access_model.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            data=dumped_data,
+            headers=self._setup_headers(dumped_data, url),
+        )
+        r.raise_for_status()
+
+
+def make_json_friendly(data):
+    if isinstance(data, set):
+        return list(sorted(data))
+    return data
```
Comment (dacamo76) [inline symbol_exporter/db_access_model.py:126]: newline

File: symbol_exporter/db_access_model.py (hunk 1 above)
Comment (CJ-Wright) [reply symbol_exporter/db_access_model.py:51]: Hmm, I was hoping to not have to pass that in to every function, would it be reasonable to move host to the init and let it be passed in but defaulted to the current value?

Comment (dacamo76) [reply symbol_exporter/db_access_model.py:51]: It's a class variable so we can access it from the class, no need for the instance - unless we are re-assigning it, in that case it should probably be an instance variable.

**Comment by CJ-Wright** [comment]
@dacamo76 ready for another round.

**Pull request merged**